### DISCUSSION TITLE
Upgrade to .NET 8 and using C# 12 feature

### DISCUSSION THREAD

**Discussion by nimanikoo** [rabbitmq/rabbitmq-dotnet-client#1527] [General]
### Is your feature request related to a problem? Please describe.

Hi 👋
I hope you are doing well.
I was reviewing the project and noticed that .NET 7 is being used for the latest version.
I also noticed some areas in the code where accessor modifiers could be applied, which might be helpful during code review.
I wanted to ask if you have any plans to upgrade to C# 12 and .NET 8?
If you are agreeable, I would be happy to help with the upgrade and submit a pull request for the changes.
I believe that upgrading to C# 12 and .NET 8 would be a positive step for the project and would make it more efficient, scalable, and future-proof.
Please let me know your thoughts on this proposal.
Thank you

### Describe the solution you'd like

Step 1: I will update all existing modules and projects to .NET 8.
Step 2: I will fix the namespaces and utilize the new features of .NET 8.
Step 3: I will perform a code review file by file to improve readability and refactor the code with references.

I am open to suggestions and feedback.
I will refactor the code to make it more modular and reusable.
I am confident that this upgrade will be a positive step for the project and I look forward to working with you on it.

Thank you for your time and consideration

**Comment by michaelklishin** [comment]
This client is very widely used and it cannot afford to require .NET 8 — that's been [out for less than two months](https://learn.microsoft.com/en-us/dotnet/core/whats-new/dotnet-8/overview) —  immediately.

At some point it will, just like the rest of the ecosystem. That can easily be a year or two away.

**Reply by nimanikoo** [reply]
Thank you for your explanation.
I was hoping to contribute and help to this project, but I understand your reasoning and it makes sense.

**Comment by lukebakken** [comment]
@nimanikoo it is unlikely this project will ever support C# language version greater than `8.0`. It will also always support `netstandard2.0` until the latter is officially unsupported by Microsoft.

**Reply by nimanikoo** [reply]
Hello  @lukebakken , I hope you're doing well.
Your words are logical and correct and we are facing a legacy code
However, if you agree, it's possible to create a new version tailored for newer editions, for example, supporting from .NET 6 or 7 onwards. My guess is that we will have much better performance in this regard, and we can gradually upgrade or rewrite various modules with the help of contributors to the newer version.
We have seen this approach in several other projects where there is a legacy version alongside a more up-to-date one.
Nevertheless, I believe C# has had significant improvements from version 8 to the current version 12 and has consistently shown growth.
Thank you again for the time you've dedicated to this project and the assistance you've provided to the open-source and .NET community.
My goal is to assist with this project, and again, your opinion is preferable in these matters.

Thank you.

**Reply by sungam3r** [reply]
> it is unlikely this project will ever support C# language version greater than 8.0. It will also always support netstandard2.0 until the latter is officially unsupported by Microsoft.

You can use any C# language version regardless of used TFMs.

**Reply by sungam3r** [reply]
LangVersion was changed to 12.0 for all projects by @danielmarbach in https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/1707.

**Reply by nimanikoo** [reply]
Thank you for informing me, I think now is the time to contribute

**Reply by danielmarbach** [reply]
This is still missing though IMHO https://github.com/rabbitmq/rabbitmq-dotnet-client/issues/1688
